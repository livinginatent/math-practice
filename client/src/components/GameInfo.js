import * as React from "react";
import Seconds from "./Seconds";
import Lives from "./Lives";
import { Container } from "@mui/system";
import { useSelector } from "react-redux";
import Points from "./Points";
import { Box } from "@mui/material";

function gameInfo() {
  return (
    <>
      <Container
        sx={{
          marginTop: 2,
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
        }}
      >
        <Box sx={{ display: "flex", flex: "1", justifyContent: "flex-start" }}>
          <Seconds></Seconds>
        </Box>
        <Box sx={{ display: "flex", justifyContent: "center" }}>
          <Points></Points>
        </Box>
        <Box sx={{ display: "flex", flex: "1", justifyContent: "flex-end" }}>
          <Lives></Lives>
        </Box>
      </Container>
    </>
  );
}

export default gameInfo;
