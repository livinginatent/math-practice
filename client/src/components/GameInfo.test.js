import { Provider,  } from "react-redux";
import GameInfo from "./GameInfo";
import configureMockStore from "redux-mock-store";
import renderer from "react-test-renderer";

describe("GameInfo component", () => {
  const initialState = {
    game: {
      points: 0,
      lives: 3,
      seconds: 20,
      isStarted: false,
      isFinished: false,
    },
  };

  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const { component } = renderer.create(
    <Provider store={store}>
      <GameInfo />
    </Provider>
  );

  test("Matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
