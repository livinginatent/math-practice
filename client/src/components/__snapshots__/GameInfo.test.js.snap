// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameInfo component Matches the snapshot 1`] = `undefined`;
